### PR TITLE
Add internal plugin truststore for validation

### PR DESCRIPTION
**Pull request by brian-mcnamara**
In response to https://youtrack.jetbrains.com/issue/IDEA-273674

Add system property to specify a truststore used for plugin validation.

### CODE REVIEW

**Comment by brian-mcnamara** [comment]
I could use some help tracking down a test class I could add a test to. Any ideas?

File: platform/platform-impl/src/com/intellij/ide/plugins/certificates/PluginCertificateStore.kt [hunk 1]
```diff
@@ -2,14 +2,46 @@
...
+    if (MANAGED_TRUSTSTORE_PATH.isNotBlank()) {
+      loadCertificates(MANAGED_TRUSTSTORE_PATH)
+    } else {
+      emptyList()
+    }
+  }
+
+  private fun loadCertificates(storePath: String) : List<X509Certificate> {
+    val keystore = KeyStore.getInstance(KeyStore.getDefaultType())
+    val keystoreFile = File(storePath)
```
Comment (AlexanderPrendota) [inline platform/platform-impl/src/com/intellij/ide/plugins/certificates/PluginCertificateStore.kt:32]: I would suggest using `Path` instead of File

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

File: platform/platform-impl/src/com/intellij/ide/plugins/certificates/PluginCertificateStore.kt [hunk 2]
```diff
@@ -2,14 +2,46 @@
...
+    if (keystoreFile.exists()) {
+      try {
+        keystore.load(FileInputStream(keystoreFile), DEFAULT_PASSWORD.toCharArray())
+        val certs = ArrayList<X509Certificate>()
+        for (alias in keystore.aliases()) {
```
Comment (AlexanderPrendota) [inline platform/platform-impl/src/com/intellij/ide/plugins/certificates/PluginCertificateStore.kt:37]:
```kotlin
keystore.aliases().mapNotNull{ keystore.getCertificate(alias) as? X509Certificate) }
```

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

**Comment by brian-mcnamara** [comment]
Its merged in to 212 branch and main https://github.com/JetBrains/intellij-community/commit/a9397aa0ce5c3964cbf5b50294b12d4447b4f01f

**Comment by AlexanderPrendota** [comment]
thanks =)

**Pull request closed without merging**